src/java/htsjdk/samtools/SRAFileReader.java

```diff
@@ -0,0 +1,275 @@
+/*===========================================================================
+*
+*                            PUBLIC DOMAIN NOTICE
+*               National Center for Biotechnology Information
+*
+*  This software/database is a "United States Government Work" under the
+*  terms of the United States Copyright Act.  It was written as part of
+*  the author's official duties as a United States Government employee and
+*  thus cannot be copyrighted.  This software/database is freely available
+*  to the public for use. The National Library of Medicine and the U.S.
+*  Government have not placed any restriction on its use or reproduction.
+*
+*  Although all reasonable efforts have been taken to ensure the accuracy
+*  and reliability of the software and data, the NLM and the U.S.
+*  Government do not and cannot warrant the performance or results that
+*  may be obtained by using this software or data. The NLM and the U.S.
+*  Government disclaim all warranties, express or implied, including
+*  warranties of performance, merchantability or fitness for any particular
+*  purpose.
+*
+*  Please cite the author in any work or product based on this material.
+*
+* ===========================================================================
+*
+*/
+
+/**
+ * Created by andrii.nikitiuk on 8/11/15.
+ */
+
+package htsjdk.samtools;
+
+import htsjdk.samtools.sra.SRAAccession;
+import htsjdk.samtools.util.CloseableIterator;
+
+import htsjdk.samtools.SamReader.Type;
+
+import htsjdk.samtools.util.Log;
+import ngs.ErrorMsg;
+import ngs.ReadCollection;
+import ngs.ReadGroupIterator;
+import ngs.ReferenceIterator;
+import ngs.Reference;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class SRAFileReader extends SamReader.ReaderImplementation implements SamReader.Indexing {
```

Comment:
Add class-level javadoc for all public classes, and method-level javadoc for all public methods (applies here and many other places).